staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -668,20 +654,16 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 		if err != nil {
 			return apierrors.NewBadRequest(fmt.Sprintf("invalid continue token: %v", err))
 		}
-		preparedKey = continueKey
+		continueKey = strings.TrimPrefix(continueKey, keyPrefix)
```

Comment:
Left the trimming and concatenation of continue key and key prefix in to avoid too many changes. This can be cleaned up in following PRs.